modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1834,6 +1853,23 @@ public CompletableFuture<TableImpl> tableAsync(int id) {
         }
     }
 
+    /**
+     * Asynchronously gets the partitions set of a table using causality token.
+     *
+     * @param causalityToken Causality token.
+     * @return Future.
+     */
+    public CompletableFuture<PartitionSet> partitionSetAsync(long causalityToken, int tableId) {
```

Comment:
Can the future return **null**?

Reply:
No, at least not by design of the versioned value